src/index.js

```diff
@@ -142,6 +142,39 @@ const updateSelectedIndex = (qualityLevels, id) => {
   });
 };
 
+// Fudge factor to account for TimeRanges rounding
+const TIME_FUDGE_FACTOR = 1 / 30;
+
+const filterRanges = function(timeRanges, predicate) {
+  let results = [];
+  let i;
+
+  if (timeRanges && timeRanges.length) {
+    // Search for ranges that match the predicate
+    for (i = 0; i < timeRanges.length; i++) {
```

Comment:
Why not use a forEach here? That would let you eliminate the `let i` above.

I think you could also use `.reduce` here.  Maybe something like:

```
let results = [];

if (timeRanges && timeRanges.length) {
  results = timeRanges.reduce((ranges, currentTimeRange, index) => {
    if (predicate(timeRanges.start(index), timeRanges.end(index)) {
      ranges.push([timeRagnes.start(index), timeRanges.end(index)]);
    }

    return ranges;
  }, []);
}

return videojs.createTime(results)
```

I'm not sure if you can do it in this project, but you could probably get rid of the timeRanges check if you're able to set a default parameter as well.

Reply:
timeRanges is just an object of `{ start(), end(), length }` so can't use things like `reduce` or `forEach` here